pkg/policy/l4.go

```diff
@@ -104,7 +120,20 @@ func CreateL4Filter(rule api.PortRule, port api.PortProtocol, direction string,
 
 		if len(l7rules) > 0 {
 			l4.L7Parser = "http"
-			l4.L7Rules = l7rules
+			if len(fromEndpoints) > 0 {
+				for _, ep := range fromEndpoints {
+					hash, err := ep.Hash()
+					if err != nil || hash == 0 {
+						log.Errorf("Could not hash (%d) endpoint %e", hash, err)
+						return L4Filter{}
+					}
+					l4.L7RulesPerEp[hash] = append(l4.L7RulesPerEp[hash], l7rules...)
+				}
+			} else if l4.Ingress {
+				// for ingress only, if there are no explicit fromEps, have a 'special' wildcard endpoint
```

Comment:
@amreshakim Why is this for ingress only?

Reply:
+ @tgraf discussed offline, seems like it may just be because egress L7Filters were not supported when this was written.